test/13_arraycommon.jl

```diff
@@ -978,6 +990,20 @@ end
                                CategoricalArray(["a", "b", "c"]))
     x[1:2] .= x[3]
     @test x == fill(get(x[3]), 3)
+
+    y = identity.(x)
+    @test x ≅ y
+    @test x !== y
+    @test y isa CategoricalArray
+
+    y = broadcast(v->v, x)
```

Comment:
maybe also test something like:
```
y = broadcast(v->x[1], x)
```
(i.e. when the cat values are produced by something else than identity transformation)

Reply:
Technically, the broadcasting machinery doesn't know that `v->v` is equivalent to `identity`, so that should test the same path, but adding a test is cheap, so why not.

Reply:
Yes - this is the same what I just did in the DataFrames.jl PR.